File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/ObjectSearchSummary.java
package com.oracle.aidataplatform.dp.model;



/**
 * A Search object in a Data Lake
* 
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ObjectSearchSummary.Builder.class)

public final class ObjectSearchSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "displayName", "description", "type", "path", "timeCreated", "timeUpdated", "owner", "createdByName", "updatedByName", "workspaceKey", "workspaceName", "catalogKey", "schemaKey", "hitHighlights"})
    public ObjectSearchSummary(String key, String displayName, String description, String type, String path, java.util.Date timeCreated, java.util.Date timeUpdated, String owner, String createdByName, String updatedByName, String workspaceKey, String workspaceName, String catalogKey, String schemaKey, java.util.List<String> hitHighlights) {
        super();
        this.key = key;
        this.displayName = displayName;
        this.description = description;
        this.type = type;
        this.path = path;
        this.timeCreated = timeCreated;
        this.timeUpdated = timeUpdated;
        this.owner = owner;
        this.createdByName = createdByName;
        this.updatedByName = updatedByName;
        this.workspaceKey = workspaceKey;
        this.workspaceName = workspaceName;
        this.catalogKey = catalogKey;
        this.schemaKey = schemaKey;
        this.hitHighlights = hitHighlights;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The key of the object
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * The key of the object
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * A user-friendly name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

        /**
         * A user-friendly name.
         * @param displayName the value to set
         * @return this builder
         **/
        

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
            /**
     * Description of the object
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * Description of the object
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
            /**
     * The type of the Object
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("type")
private String type;

        /**
         * The type of the Object
         * @param type the value to set
         * @return this builder
         **/
        

public Builder type(String type) {
    this.type = type;
    return this;
}
            /**
     * Path of object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("path")
private String path;

        /**
         * Path of object.
         * @param path the value to set
         * @return this builder
         **/
        

public Builder path(String path) {
    this.path = path;
    return this;
}
            /**
     * The date and time the object was created, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * The date and time the object was created, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * The date and time the object was updated, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

        /**
         * The date and time the object was updated, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
         * @param timeUpdated the value to set
         * @return this builder
         **/
        

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
            /**
     * Name of the owner of the record
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("owner")
private String owner;

        /**
         * Name of the owner of the record
         * @param owner the value to set
         * @return this builder
         **/
        

public Builder owner(String owner) {
    this.owner = owner;
    return this;
}
            /**
     * Name of the user who created this record
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("createdByName")
private String createdByName;

        /**
         * Name of the user who created this record
         * @param createdByName the value to set
         * @return this builder
         **/
        

public Builder createdByName(String createdByName) {
    this.createdByName = createdByName;
    return this;
}
            /**
     * Name of the user who updated this record
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("updatedByName")
private String updatedByName;

        /**
         * Name of the user who updated this record
         * @param updatedByName the value to set
         * @return this builder
         **/
        

public Builder updatedByName(String updatedByName) {
    this.updatedByName = updatedByName;
    return this;
}
            /**
     * Key of the workspace.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("workspaceKey")
private String workspaceKey;

        /**
         * Key of the workspace.
         * @param workspaceKey the value to set
         * @return this builder
         **/
        

public Builder workspaceKey(String workspaceKey) {
    this.workspaceKey = workspaceKey;
    return this;
}
            /**
     * Name of the workspace.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("workspaceName")
private String workspaceName;

        /**
         * Name of the workspace.
         * @param workspaceName the value to set
         * @return this builder
         **/
        

public Builder workspaceName(String workspaceName) {
    this.workspaceName = workspaceName;
    return this;
}
            /**
     * Key of the catalog.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("catalogKey")
private String catalogKey;

        /**
         * Key of the catalog.
         * @param catalogKey the value to set
         * @return this builder
         **/
        

public Builder catalogKey(String catalogKey) {
    this.catalogKey = catalogKey;
    return this;
}
            /**
     * Key of the schema.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("schemaKey")
private String schemaKey;

        /**
         * Key of the schema.
         * @param schemaKey the value to set
         * @return this builder
         **/
        

public Builder schemaKey(String schemaKey) {
    this.schemaKey = schemaKey;
    return this;
}
            /**
     * Highlights related to this notebook object
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("hitHighlights")
private java.util.List<String> hitHighlights;

        /**
         * Highlights related to this notebook object
         * @param hitHighlights the value to set
         * @return this builder
         **/
        

public Builder hitHighlights(java.util.List<String> hitHighlights) {
    this.hitHighlights = hitHighlights;
    return this;
}


        public ObjectSearchSummary build() {
            ObjectSearchSummary model = new ObjectSearchSummary(this.key
                , this.displayName
                , this.description
                , this.type
                , this.path
                , this.timeCreated
                , this.timeUpdated
                , this.owner
                , this.createdByName
                , this.updatedByName
                , this.workspaceKey
                , this.workspaceName
                , this.catalogKey
                , this.schemaKey
                , this.hitHighlights);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ObjectSearchSummary model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.description(model.getDescription());
    this.type(model.getType());
    this.path(model.getPath());
    this.timeCreated(model.getTimeCreated());
    this.timeUpdated(model.getTimeUpdated());
    this.owner(model.getOwner());
    this.createdByName(model.getCreatedByName());
    this.updatedByName(model.getUpdatedByName());
    this.workspaceKey(model.getWorkspaceKey());
    this.workspaceName(model.getWorkspaceName());
    this.catalogKey(model.getCatalogKey());
    this.schemaKey(model.getSchemaKey());
    this.hitHighlights(model.getHitHighlights());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The key of the object
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * The key of the object
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * A user-friendly name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * A user-friendly name.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * Description of the object
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * Description of the object
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * The type of the Object
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("type")
    private final String type;

        /**
     * The type of the Object
     * @return the value
     **/
    
    public String getType() {
        return type;
    }


        /**
     * Path of object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("path")
    private final String path;

        /**
     * Path of object.
     * @return the value
     **/
    
    public String getPath() {
        return path;
    }


        /**
     * The date and time the object was created, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * The date and time the object was created, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * The date and time the object was updated, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
    private final java.util.Date timeUpdated;

        /**
     * The date and time the object was updated, in the format defined by [RFC 3339](https://tools.ietf.org/html/rfc3339).
* Example: {@code 2016-08-25T21:10:29.600Z}
* 
     * @return the value
     **/
    
    public java.util.Date getTimeUpdated() {
        return timeUpdated;
    }


        /**
     * Name of the owner of the record
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("owner")
    private final String owner;

        /**
     * Name of the owner of the record
     * @return the value
     **/
    
    public String getOwner() {
        return owner;
    }


        /**
     * Name of the user who created this record
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("createdByName")
    private final String createdByName;

        /**
     * Name of the user who created this record
     * @return the value
     **/
    
    public String getCreatedByName() {
        return createdByName;
    }


        /**
     * Name of the user who updated this record
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("updatedByName")
    private final String updatedByName;

        /**
     * Name of the user who updated this record
     * @return the value
     **/
    
    public String getUpdatedByName() {
        return updatedByName;
    }


        /**
     * Key of the workspace.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("workspaceKey")
    private final String workspaceKey;

        /**
     * Key of the workspace.
     * @return the value
     **/
    
    public String getWorkspaceKey() {
        return workspaceKey;
    }


        /**
     * Name of the workspace.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("workspaceName")
    private final String workspaceName;

        /**
     * Name of the workspace.
     * @return the value
     **/
    
    public String getWorkspaceName() {
        return workspaceName;
    }


        /**
     * Key of the catalog.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("catalogKey")
    private final String catalogKey;

        /**
     * Key of the catalog.
     * @return the value
     **/
    
    public String getCatalogKey() {
        return catalogKey;
    }


        /**
     * Key of the schema.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("schemaKey")
    private final String schemaKey;

        /**
     * Key of the schema.
     * @return the value
     **/
    
    public String getSchemaKey() {
        return schemaKey;
    }


        /**
     * Highlights related to this notebook object
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("hitHighlights")
    private final java.util.List<String> hitHighlights;

        /**
     * Highlights related to this notebook object
     * @return the value
     **/
    
    public java.util.List<String> getHitHighlights() {
        return hitHighlights;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ObjectSearchSummary(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", type=").append(String.valueOf(this.type));
        sb.append(", path=").append(String.valueOf(this.path));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", timeUpdated=").append(String.valueOf(this.timeUpdated));
        sb.append(", owner=").append(String.valueOf(this.owner));
        sb.append(", createdByName=").append(String.valueOf(this.createdByName));
        sb.append(", updatedByName=").append(String.valueOf(this.updatedByName));
        sb.append(", workspaceKey=").append(String.valueOf(this.workspaceKey));
        sb.append(", workspaceName=").append(String.valueOf(this.workspaceName));
        sb.append(", catalogKey=").append(String.valueOf(this.catalogKey));
        sb.append(", schemaKey=").append(String.valueOf(this.schemaKey));
        sb.append(", hitHighlights=").append(String.valueOf(this.hitHighlights));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ObjectSearchSummary)) {
            return false;
        }

        ObjectSearchSummary other = (ObjectSearchSummary) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.type, other.type) &&
            java.util.Objects.equals(this.path, other.path) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.timeUpdated, other.timeUpdated) &&
            java.util.Objects.equals(this.owner, other.owner) &&
            java.util.Objects.equals(this.createdByName, other.createdByName) &&
            java.util.Objects.equals(this.updatedByName, other.updatedByName) &&
            java.util.Objects.equals(this.workspaceKey, other.workspaceKey) &&
            java.util.Objects.equals(this.workspaceName, other.workspaceName) &&
            java.util.Objects.equals(this.catalogKey, other.catalogKey) &&
            java.util.Objects.equals(this.schemaKey, other.schemaKey) &&
            java.util.Objects.equals(this.hitHighlights, other.hitHighlights);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.type == null ? 43 : this.type.hashCode());
        result = (result * PRIME) + (this.path == null ? 43 : this.path.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.timeUpdated == null ? 43 : this.timeUpdated.hashCode());
        result = (result * PRIME) + (this.owner == null ? 43 : this.owner.hashCode());
        result = (result * PRIME) + (this.createdByName == null ? 43 : this.createdByName.hashCode());
        result = (result * PRIME) + (this.updatedByName == null ? 43 : this.updatedByName.hashCode());
        result = (result * PRIME) + (this.workspaceKey == null ? 43 : this.workspaceKey.hashCode());
        result = (result * PRIME) + (this.workspaceName == null ? 43 : this.workspaceName.hashCode());
        result = (result * PRIME) + (this.catalogKey == null ? 43 : this.catalogKey.hashCode());
        result = (result * PRIME) + (this.schemaKey == null ? 43 : this.schemaKey.hashCode());
        result = (result * PRIME) + (this.hitHighlights == null ? 43 : this.hitHighlights.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/object_search_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ObjectSearchSummary(object):
    """
    A Search object in a Data Lake
    """

    def __init__(self, **kwargs):
        """
        Initializes a new ObjectSearchSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this ObjectSearchSummary.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this ObjectSearchSummary.
        :type display_name: str

        :param description:
            The value to assign to the description property of this ObjectSearchSummary.
        :type description: str

        :param type:
            The value to assign to the type property of this ObjectSearchSummary.
        :type type: str

        :param path:
            The value to assign to the path property of this ObjectSearchSummary.
        :type path: str

        :param time_created:
            The value to assign to the time_created property of this ObjectSearchSummary.
        :type time_created: datetime

        :param time_updated:
            The value to assign to the time_updated property of this ObjectSearchSummary.
        :type time_updated: datetime

        :param owner:
            The value to assign to the owner property of this ObjectSearchSummary.
        :type owner: str

        :param created_by_name:
            The value to assign to the created_by_name property of this ObjectSearchSummary.
        :type created_by_name: str

        :param updated_by_name:
            The value to assign to the updated_by_name property of this ObjectSearchSummary.
        :type updated_by_name: str

        :param workspace_key:
            The value to assign to the workspace_key property of this ObjectSearchSummary.
        :type workspace_key: str

        :param workspace_name:
            The value to assign to the workspace_name property of this ObjectSearchSummary.
        :type workspace_name: str

        :param catalog_key:
            The value to assign to the catalog_key property of this ObjectSearchSummary.
        :type catalog_key: str

        :param schema_key:
            The value to assign to the schema_key property of this ObjectSearchSummary.
        :type schema_key: str

        :param hit_highlights:
            The value to assign to the hit_highlights property of this ObjectSearchSummary.
        :type hit_highlights: list[str]

        """
        self.swagger_types = {
            'key': 'str',
            'display_name': 'str',
            'description': 'str',
            'type': 'str',
            'path': 'str',
            'time_created': 'datetime',
            'time_updated': 'datetime',
            'owner': 'str',
            'created_by_name': 'str',
            'updated_by_name': 'str',
            'workspace_key': 'str',
            'workspace_name': 'str',
            'catalog_key': 'str',
            'schema_key': 'str',
            'hit_highlights': 'list[str]'
        }

        self.attribute_map = {
            'key': 'key',
            'display_name': 'displayName',
            'description': 'description',
            'type': 'type',
            'path': 'path',
            'time_created': 'timeCreated',
            'time_updated': 'timeUpdated',
            'owner': 'owner',
            'created_by_name': 'createdByName',
            'updated_by_name': 'updatedByName',
            'workspace_key': 'workspaceKey',
            'workspace_name': 'workspaceName',
            'catalog_key': 'catalogKey',
            'schema_key': 'schemaKey',
            'hit_highlights': 'hitHighlights'
        }

        self._key = None
        self._display_name = None
        self._description = None
        self._type = None
        self._path = None
        self._time_created = None
        self._time_updated = None
        self._owner = None
        self._created_by_name = None
        self._updated_by_name = None
        self._workspace_key = None
        self._workspace_name = None
        self._catalog_key = None
        self._schema_key = None
        self._hit_highlights = None

    @property
    def key(self):
        """
        **[Required]** Gets the key of this ObjectSearchSummary.
        The key of the object


        :return: The key of this ObjectSearchSummary.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this ObjectSearchSummary.
        The key of the object


        :param key: The key of this ObjectSearchSummary.
        :type: str
        """
        self._key = key

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this ObjectSearchSummary.
        A user-friendly name.


        :return: The display_name of this ObjectSearchSummary.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this ObjectSearchSummary.
        A user-friendly name.


        :param display_name: The display_name of this ObjectSearchSummary.
        :type: str
        """
        self._display_name = display_name

    @property
    def description(self):
        """
        **[Required]** Gets the description of this ObjectSearchSummary.
        Description of the object


        :return: The description of this ObjectSearchSummary.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this ObjectSearchSummary.
        Description of the object


        :param description: The description of this ObjectSearchSummary.
        :type: str
        """
        self._description = description

    @property
    def type(self):
        """
        **[Required]** Gets the type of this ObjectSearchSummary.
        The type of the Object


        :return: The type of this ObjectSearchSummary.
        :rtype: str
        """
        return self._type

    @type.setter
    def type(self, type):
        """
        Sets the type of this ObjectSearchSummary.
        The type of the Object


        :param type: The type of this ObjectSearchSummary.
        :type: str
        """
        self._type = type

    @property
    def path(self):
        """
        **[Required]** Gets the path of this ObjectSearchSummary.
        Path of object.


        :return: The path of this ObjectSearchSummary.
        :rtype: str
        """
        return self._path

    @path.setter
    def path(self, path):
        """
        Sets the path of this ObjectSearchSummary.
        Path of object.


        :param path: The path of this ObjectSearchSummary.
        :type: str
        """
        self._path = path

    @property
    def time_created(self):
        """
        Gets the time_created of this ObjectSearchSummary.
        The date and time the object was created, in the format defined by `RFC 3339`__.
        Example: `2016-08-25T21:10:29.600Z`

        __ https://tools.ietf.org/html/rfc3339


        :return: The time_created of this ObjectSearchSummary.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this ObjectSearchSummary.
        The date and time the object was created, in the format defined by `RFC 3339`__.
        Example: `2016-08-25T21:10:29.600Z`

        __ https://tools.ietf.org/html/rfc3339


        :param time_created: The time_created of this ObjectSearchSummary.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def time_updated(self):
        """
        **[Required]** Gets the time_updated of this ObjectSearchSummary.
        The date and time the object was updated, in the format defined by `RFC 3339`__.
        Example: `2016-08-25T21:10:29.600Z`

        __ https://tools.ietf.org/html/rfc3339


        :return: The time_updated of this ObjectSearchSummary.
        :rtype: datetime
        """
        return self._time_updated

    @time_updated.setter
    def time_updated(self, time_updated):
        """
        Sets the time_updated of this ObjectSearchSummary.
        The date and time the object was updated, in the format defined by `RFC 3339`__.
        Example: `2016-08-25T21:10:29.600Z`

        __ https://tools.ietf.org/html/rfc3339


        :param time_updated: The time_updated of this ObjectSearchSummary.
        :type: datetime
        """
        self._time_updated = time_updated

    @property
    def owner(self):
        """
        Gets the owner of this ObjectSearchSummary.
        Name of the owner of the record


        :return: The owner of this ObjectSearchSummary.
        :rtype: str
        """
        return self._owner

    @owner.setter
    def owner(self, owner):
        """
        Sets the owner of this ObjectSearchSummary.
        Name of the owner of the record


        :param owner: The owner of this ObjectSearchSummary.
        :type: str
        """
        self._owner = owner

    @property
    def created_by_name(self):
        """
        **[Required]** Gets the created_by_name of this ObjectSearchSummary.
        Name of the user who created this record


        :return: The created_by_name of this ObjectSearchSummary.
        :rtype: str
        """
        return self._created_by_name

    @created_by_name.setter
    def created_by_name(self, created_by_name):
        """
        Sets the created_by_name of this ObjectSearchSummary.
        Name of the user who created this record


        :param created_by_name: The created_by_name of this ObjectSearchSummary.
        :type: str
        """
        self._created_by_name = created_by_name

    @property
    def updated_by_name(self):
        """
        Gets the updated_by_name of this ObjectSearchSummary.
        Name of the user who updated this record


        :return: The updated_by_name of this ObjectSearchSummary.
        :rtype: str
        """
        return self._updated_by_name

    @updated_by_name.setter
    def updated_by_name(self, updated_by_name):
        """
        Sets the updated_by_name of this ObjectSearchSummary.
        Name of the user who updated this record


        :param updated_by_name: The updated_by_name of this ObjectSearchSummary.
        :type: str
        """
        self._updated_by_name = updated_by_name

    @property
    def workspace_key(self):
        """
        Gets the workspace_key of this ObjectSearchSummary.
        Key of the workspace.


        :return: The workspace_key of this ObjectSearchSummary.
        :rtype: str
        """
        return self._workspace_key

    @workspace_key.setter
    def workspace_key(self, workspace_key):
        """
        Sets the workspace_key of this ObjectSearchSummary.
        Key of the workspace.


        :param workspace_key: The workspace_key of this ObjectSearchSummary.
        :type: str
        """
        self._workspace_key = workspace_key

    @property
    def workspace_name(self):
        """
        Gets the workspace_name of this ObjectSearchSummary.
        Name of the workspace.


        :return: The workspace_name of this ObjectSearchSummary.
        :rtype: str
        """
        return self._workspace_name

    @workspace_name.setter
    def workspace_name(self, workspace_name):
        """
        Sets the workspace_name of this ObjectSearchSummary.
        Name of the workspace.


        :param workspace_name: The workspace_name of this ObjectSearchSummary.
        :type: str
        """
        self._workspace_name = workspace_name

    @property
    def catalog_key(self):
        """
        Gets the catalog_key of this ObjectSearchSummary.
        Key of the catalog.


        :return: The catalog_key of this ObjectSearchSummary.
        :rtype: str
        """
        return self._catalog_key

    @catalog_key.setter
    def catalog_key(self, catalog_key):
        """
        Sets the catalog_key of this ObjectSearchSummary.
        Key of the catalog.


        :param catalog_key: The catalog_key of this ObjectSearchSummary.
        :type: str
        """
        self._catalog_key = catalog_key

    @property
    def schema_key(self):
        """
        Gets the schema_key of this ObjectSearchSummary.
        Key of the schema.


        :return: The schema_key of this ObjectSearchSummary.
        :rtype: str
        """
        return self._schema_key

    @schema_key.setter
    def schema_key(self, schema_key):
        """
        Sets the schema_key of this ObjectSearchSummary.
        Key of the schema.


        :param schema_key: The schema_key of this ObjectSearchSummary.
        :type: str
        """
        self._schema_key = schema_key

    @property
    def hit_highlights(self):
        """
        Gets the hit_highlights of this ObjectSearchSummary.
        Highlights related to this notebook object


        :return: The hit_highlights of this ObjectSearchSummary.
        :rtype: list[str]
        """
        return self._hit_highlights

    @hit_highlights.setter
    def hit_highlights(self, hit_highlights):
        """
        Sets the hit_highlights of this ObjectSearchSummary.
        Highlights related to this notebook object


        :param hit_highlights: The hit_highlights of this ObjectSearchSummary.
        :type: list[str]
        """
        self._hit_highlights = hit_highlights

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
